src/Juvix/Compiler/Core/Language/Nodes.hs

```diff
@@ -110,6 +110,7 @@ data Pattern' i
   = PatWildcard (PatternWildcard' i)
   | PatBinder (PatternBinder' i)
   | PatConstr (PatternConstr' i)
+  | PatConst (PatternConst' i)
```

Comment:
I think at the level of a complex Match it might make sense to also just have a "PatSucc" pattern constructor. Then when compiling pattern matching to Cases, one would get rid of PatConst and PatSucc in favour of builtin comparison with 0 and case on the resulting boolean.

Reply:
But already with the complex Match nats should be translated to integers, because otherwise you can't evaluate arithmetic operations.